File: Python/BOJ/13305 주유소.py
import sys
N = int(sys.stdin.readline())
line = list(map(int, sys.stdin.readline().split()))
price = list(map(int, sys.stdin.readline().split()))


a = line[0]*price[0]
b = price[0]

for i in range(1, N-1):
    if b > price[i]:
        b = price[i]
    a += b*line[i]
print(a)

File: Java/BOJ/13305 주유소.java
import java.io.*;
import java.util.*;

public class Main {    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        long[] line = new long[N-1];
        for (int i = 0; i < N-1; i ++) {
        	line[i] = Integer.parseInt(st.nextToken());
        }
        long[] price = new long[N];
        st = new StringTokenizer(br.readLine(), " ");
        for(int i = 0; i < N; i++) {
        	price[i] = Integer.parseInt(st.nextToken());
        }
        
        long a = line[0] * price[0];
        long b = price[0];
        
        for (int i = 1; i < N-1; i++) {
        	if (b > price[i]) {
        		b = price[i];
        	}
        	a += b *line[i];
        }
        System.out.println(a);
    }
}